src/gowrapper/ckks/keygen.go

```diff
@@ -14,37 +17,44 @@ struct Lattigo_KeyPairHandle {
 import "C"
 
 import (
+	"errors"
 	"github.com/ldsec/lattigo/v2/ckks"
+	"github.com/ldsec/lattigo/v2/rlwe"
```

Comment:
This is part of the Lattigo (dependency) refactor: they moved a bunch of functionality to their `rlwe` module so that it could be reused between multiple cryptosystems implemented in the library. RLWE = ring-LWE = ring learning with errors, a mathematical problem on which most efficient lattice cryptosystems are based. Thus by moving functionality to their `rlwe` module, they can reuse it across many implementations.

Reply:
Makes sense to me. I am (slightly) familiar with ring-LWE but didn't (don't) know if it is related to CKKS (since I'm not that familiar with CKKS yet) and so didn't know if it made sense for them to be sharing logic. This question was mostly to ensure there wasn't an accidental refactor or other major change we didn't intend.

Reply:
The right way to think about CKKS is that it is a "wrapper" around a generic homomorphic encryption scheme. The original paper is just a modification of BGV-style HE, but it points out that it can also be applied to other HE styles (all of which use LWE or a variant thereof).